src/pint/toa.py

```diff
@@ -1888,6 +1895,16 @@ def compute_pulse_numbers(self, model):
         self.table["pulse_number"] = phases.int
         self.table["pulse_number"].unit = u.dimensionless_unscaled
 
+    def remove_pulse_numbers(self):
+        """Delete pulse numbers from TOA data"""
+
+        if "pulse_number" in self.table.colnames:
+            del self.table["pulse_number"]
+        else:
+            log.warning(
```

Comment:
I'd rather see this as "delete pulse numbers if present" and have it run smoothly with no warnings if they're not present - then this is a cheap way to make sure there are no pulse numbers (just before saving, say).

Reply:
ok